### DISCUSSION TITLE
Trouble with SSH

### DISCUSSION THREAD

**Discussion by southkonafarms** [community/community#21577] [New to GitHub]
<p>I’m trying to use SSL instead of a password.  It worked one time but now fails.  This is the message I receive:<br>
ssh -T <a href="mailto:git@github.com">git@github.com</a><br>
Hi southkonafarms! You’ve successfully authenticated, but GitHub does not provide shell access.</p>
<p>???</p>
<p>-south kona farms</p>

**Comment by airtower-luna** [comment]
<p>That is no failure. It means your SSH key works and GitHub recognizes it for your account. You can now use SSH as a transport for Git (e.g. <code>git push</code>).</p>

**Comment by southkonafarms** [comment]
<p>When I type <em>git push</em>, it continues to ask for a password.  What am I doing wrong?</p>
<ul>
<li>south kona farms</li>
</ul>

**Comment by airtower-luna** [comment]
<p>Most likely your local repository is still configured to use a HTTPS URL to access the GitHub repository. You need to change the URL to the SSH one. You can find the right URL by clicking the green “Code” button on the repository page and selecting “SSH”, it should look something like this:</p>
<pre><code class="lang-nohighlight">git@github.com:user/repo.git
</code></pre>
<p>You can change the remote URL with the following command (change URL and remote name as needed):</p>
<pre><code class="lang-nohighlight">git remote set-url origin git@github.com:user/repo.git
</code></pre>
<p><code>origin</code> is the default remote name for where you initially cloned from. If you need to use something different, you probably already know.</p>
<p>See also: <a href="https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes" class="inline-onebox">Git - Working with Remotes</a></p>

**Comment by southkonafarms** [comment, marked as answer]
<p>Thank you</p>
<p>git remote set-url origin <a href="mailto:git@github.com">git@github.com</a>:southkonafarms/cosmos.git</p>
<p>did the trick.</p>
<p>south kona farms</p>

**Comment by jsoref** [comment]
<p>Please select <a href="https://github.community/t/trouble-with-ssl/183938/4" class="inline-onebox">Trouble with SSL - #4 by airtower-luna</a> and mark it as the solution.</p>

**Comment by southkonafarms** [comment]
<p>Solution posted<br>
git remote set-url origin <a href="mailto:git@github.com">git@github.com</a>:southkonafarms/cosmos.git</p>
